data_diff/database.py

```diff
@@ -186,6 +287,12 @@ def md5_to_int(self, s: str) -> str:
     def to_string(self, s: str):
         return f"{s}::varchar"
 
+    def normalize_value_by_type(self, value, coltype: ColType) -> str:
+        if isinstance(coltype, TemporalType):
+            timestamp = self.to_string(f"{value}::timestamp({coltype.precision})")
+            return f"RPAD({timestamp}, {TIMESTAMP_BASE_LEN + coltype.precision}, '0')"
```

Comment:
So MySQL below does right-padding by default to the precision? 👍🏻 

FWIW, I think in the abstract PyDoc (since it's where people will look when implementing a new driver) for `#normalize_value_by_type` I think it's worth explaining in detail how we convert everything to a string before checksumming (to avoid relying on e.g. different implementations of floats, varints, ...). 

Here, I'd also start the list of types that we expect some canonical representation for, in this case, with Datetime/Timestamp we always expect: `2022-06-03 12:24:35.000000` (if precision is 6).